Add width for codeblocks in comments

sec65 opened the pull request:
Resolves #5359

Adds a new doc_comment_code_block_width configuration (default: 100), which limits the code blocks in comments, if format_code_in_doc_comments is true.

calebcartwright commented:
Thank you for this! I'm in the process of prepping the next release but will circle back to this afterwards

calebcartwright commented on Configurations.md:
```diff
@@ -926,6 +926,14 @@ fn add_one(x: i32) -> i32 {
+- **Default value**: `100`
+- **Possible values**: any positive integer that is less than or equal to the value specified for [`max_width`](#max_width)
+- **Stable**: No (tracking issue: [#5359](https://github.com/rust-lang/rustfmt/issues/5359))
+
```
I think it would be good to include a link to the `format_code_in_doc_comments` option, similar to what's done for other options https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#array_width

What do you think?

calebcartwright commented on tests/source/issue-5359/code_comment_width_greater_max_width.rs:
```diff
@@ -0,0 +1,17 @@
+// rustfmt-max_width: 50
```
These tests look good! However, we actually put tests for config option in a special directory, named `configs` under the source and target dirs

Please move these when you get a chance, and with that move you may be able to rename the files (though don't feel obligated to do so)
https://github.com/rust-lang/rustfmt/tree/master/tests/source/configs

calebcartwright reviewed (commented):
Thanks this looks great! Couple small items but should be good to merge afterwards :rocket:

sec65 commented:
@calebcartwright Thanks for your review. I changed the docu and moved the tests to the config folder and renamed the files to mach with the tested option.

Short question: In one of the test cases I wanted to use (and by that test) the default value, but there seems to be a check in the configs folder, that the config option has to be set explicitily (so I did that now). Or is there a way around it?

calebcartwright commented:
Good question! It's actually better to have the options explicitly set. For these tests we're just trying to validate the formatting behavior when options have a certain value and not so concerned with using these to validate defaults.

Once an option is stabilized we can't change the default and while options remain unstable there's still a chance that we could change the defaults, all of which makes it better for the system & idempotence tests to be explicit

calebcartwright commented:
Everything looks good to me, thanks you and nice first contribution!
